base/compiler/ssair/driver.jl

```diff
@@ -93,6 +94,9 @@ function convert_to_ircode(ci::CodeInfo, code::Vector{Any}, coverage::Bool, narg
                 inbounds_depth -= 1
             end
             stmt = nothing
+        elseif isexpr(stmt, :noinline)
+            disable_inline = true
```

Comment:
Why the need for `disable_inline`? We should be able to set the `IR_FLAG_NOINLINE` directly.

Reply:
The macro currently expands like this:
```Julia
julia> f(x) = x
f (generic function with 1 method)

julia> @macroexpand @noinline f(3)
quote
    $(Expr(:noinline))
    f(3)
end
```
After the `:noinline` expression is read, we need to record `disable_inline` so that the next iteration of the loop can set the flag(The iteration of the loop in which the `:noinline` expression is evaluated skips the flag setting because it sets `stmt` to `nothing`). The idea is the expression after the `:noinline` statement is the one that gets the flag set, rather than the `:noinline` expression itself. So that is why we need to record `disable_inline`.